std-bits/table/src/main/java/org/enso/table/operations/RunningIteratorFactory.java

```diff
@@ -3,4 +3,6 @@
 public interface RunningIteratorFactory {
 
   RunningIterator getIterator();
+
+  void calculateNextValue(int i, RunningIterator it);
```

Comment:
It feels a bit weird to me that the **Factor** has this method, it felt more fitting in the Generator before.

I understand it needs some more context info? Maybe it could be moved into `RunningIterator` itself?

I'm not sure but it feels out of place for the _Factory_ to be performing such calculations.

Reply:
Ack. Sorry managed to pick up the start of the next refactor. I promise it will make more sense when it is finished.

Reply:
I'm going to attempt to back that part out again.

Reply:
OK. It's gone. Sorry about that.

Reply:
You made me curious then :)